From 4ce0b451b784eb37f7d517784760c9a7874d81c3 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 29 Jul 2024 18:30:10 +0200
Subject: [PATCH 1042/1695] clk: stm32mp25: do not reset flexgen crossbar input
 if not TDCID

Let the TDCID handle the crossbar channel input selection default value.
Otherwise, if authorized, the CPU can modify root clocks input signals
resulting on severe impacts on all the software ecosystem.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 18174d6d09e723ff2b1cfb1ca127cacc0f1e53cc clk: stm32mp25: do not reset flexgen crossbar input if not TDCID]
Change-Id: Ib05fd2c221112a500f4fb9dcc7b37c0f6ec97036
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/395776
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 44f13ca65..a84f52d2c 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -1674,9 +1674,14 @@ static void stm32mp2_clk_xbar_on_hsi(struct clk_stm32_priv *priv)
 	uintptr_t xbar0cfgr = priv->base + RCC_XBAR0CFGR;
 	uint32_t i = 0;
 
-	for (i = 0; i < XBAR_ROOT_CHANNEL_NB; i++)
+	if (IS_ENABLED(CFG_STM32_CM33TDCID))
+		return;
+
+	for (i = 0; i < XBAR_ROOT_CHANNEL_NB; i++) {
 		io_clrsetbits32(xbar0cfgr + (0x4 * i),
 				RCC_XBAR0CFGR_XBAR0SEL_MASK, XBAR_SRC_HSI);
+	}
+
 }
 
 static int stm32mp2_a35_pll1_start(void)
-- 
2.47.2

